ons-spark/raw-notebooks/reading-and-writing-data/reading-and-writing-data-spark.ipynb

```diff
@@ -0,0 +1,609 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Reading and Writing Data in Spark\n",
+    "This chapter will go into more detail about the various file formats available to use with Spark, and how Spark interacts with these file formats. [Introduction to PySpark](../pyspark-intro/pyspark-intro) and [Introduction to SparklyR](../sparklyr-intro) briefly covered CSV files and Parquet files and some basic differences between them.\n",
+    "\n",
+    "This chapter will provide more detail on parquet files, CSV files, ORC files and Avro files, the differences between them and how to read and write data using these formats.\n",
+    "\n",
+    "Let's start by setting up a Spark session and reading in the config file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pyspark.sql import SparkSession, functions as F\n",
+    "import yaml\n",
+    "\n",
+    "spark = (SparkSession.builder.master(\"local[2]\")\n",
+    "         .appName(\"reading-data\")\n",
+    "         .getOrCreate())\n",
+    "\n",
+    "with open(\"../../../config.yaml\") as f:\n",
+    "    config = yaml.safe_load(f)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```r\n",
+    "library(sparklyr)\n",
+    "library(dplyr)\n",
+    "\n",
+    "sc <- sparklyr::spark_connect(\n",
+    "  master = \"local[2]\",\n",
+    "  app_name = \"ons-spark\",\n",
+    "  config = list(default = list(spark.submit.deployMode = \"cluster\")),\n",
```

Comment:
Can't remember why I set the config parameter on spark_connect like this but probably better to change it to config = sparklyr::spark_config() to be consistent with the rest of the book

Reply:
Good catch, I've amended that and checked and the book is displaying correctly with those changes.